### PR TITLE
[BugFix] Column mapping of table function

### LINKED ISSUE #25218: [Simple load from S3] execute create table as select * from table function, be crash
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
create database starrocks_insert_test_db0fdd2c_0a7a_11ee_b9e3_00163e0e550b
use starrocks_insert_test_db0fdd2c_0a7a_11ee_b9e3_00163e0e550b
 create table ctas1 as select * from table("path" = "s3://xxx/xxx/xxx/1w_rows",
                          "format" = "parquet",
                          "aws.s3.use_instance_profile" = "false",
                          "aws.s3.region" = "xxx",
                          "aws.s3.access_key"= "xxx",
                          "aws.s3.secret_key"="xxx"
                           );
```
*** Aborted at 1686723382 (unix time) try "date -d @1686723382" if you are using GNU date ***
PC: @     0x7f3eadce2589 __memcpy_ssse3_back
*** SIGSEGV (@0x7f3dad4f3000) received by PID 19681 (TID 0x7f3e47f4d700) from PID 18446744072322232320; stack trace: ***
    @          0x5f382e2 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7f3eae677630 (unknown)
    @     0x7f3eadce2589 __memcpy_ssse3_back
    @          0x519de38 starrocks::stream_load::OlapTableSink::_print_varchar_error_msg()
    @          0x51a18c8 starrocks::stream_load::OlapTableSink::_validate_data()
    @          0x51b08d3 starrocks::stream_load::OlapTableSink::send_chunk()
    @          0x5223b9b starrocks::pipeline::OlapTableSinkOperator::push_chunk()
    @          0x2874771 starrocks::pipeline::PipelineDriver::process()
    @          0x523a2cf starrocks::pipeline::GlobalDriverExecutor::_worker_thread()
    @          0x4b5dbb2 starrocks::ThreadPool::dispatch_thread()
    @          0x4b586aa starrocks::Thread::supervise_thread()
    @     0x7f3eae66fea5 start_thread
    @     0x7f3eadc8ab0d __clone
    @                0x0 (unknown)

### Expected behavior (Required)
execute successfuly
### Real behavior (Required)
be crash

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
mysql> select current_version();
+--------------------+
| current_version()  |
+--------------------+
| 3.1.0-0614-953a8a9 |
+--------------------+
1 row in set (0.05 sec)

### PR DESCRIPTION
**Pull request by rickif**
Fixes https://github.com/StarRocks/starrocks/issues/25218 #25288 #25358 #25275 #25285

This PR fixes the column mapping issue.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/service/internal_service.cpp [hunk 1]
```diff
@@ -808,6 +808,7 @@ void PInternalServiceImplBase<T>::get_file_schema(google::protobuf::RpcControlle
 
     std::vector<SlotDescriptor> schema;
     st = p_scanner->get_schema(&schema);
+    LOG_IF(WARNING, !st.ok()) << "get schema failed: " << st;
```
Comment (meegoo) [inline be/src/service/internal_service.cpp:811]: should return?

Comment (rickif) [reply be/src/service/internal_service.cpp]: Updated.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 5 (20.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|2|00.00%|[3107, 3131]|
|:large_blue_circle:|com/starrocks/analysis/Analyzer.java|1|3|33.33%|[339, 340]|

File: be/src/service/internal_service.cpp [hunk 2]
```diff
@@ -808,6 +808,10 @@ void PInternalServiceImplBase<T>::get_file_schema(google::protobuf::RpcControlle
 
     std::vector<SlotDescriptor> schema;
     st = p_scanner->get_schema(&schema);
+    if (!st.ok()) {
```
Comment (chaoyli) [inline be/src/service/internal_service.cpp:811]: Output a more concise message in the log.

Comment (rickif) [reply be/src/service/internal_service.cpp:811]: Updated.

**Comment by rickif** [comment]
@mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 5 (20.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|2|00.00%|[3109, 3133]|
|:large_blue_circle:|com/starrocks/analysis/Analyzer.java|1|3|33.33%|[339, 340]|

File: be/src/service/internal_service.cpp [hunk 3]
```diff
@@ -793,21 +793,30 @@ void PInternalServiceImplBase<T>::get_file_schema(google::protobuf::RpcControlle
...
+        st = Status::InvalidArgument(err_msg);
         return;
     }
 
     DeferOp defer2([&p_scanner] { p_scanner->close(); });
 
     std::vector<SlotDescriptor> schema;
     st = p_scanner->get_schema(&schema);
+    if (!st.ok()) {
+        auto err_msg = fmt::format("get file schema failed: {}, file path: {}", st, scan_range.ranges[0].path);
```
Comment (meegoo) [inline be/src/service/internal_service.cpp:817]: err_msg unused?

Comment (rickif) [reply be/src/service/internal_service.cpp:817]: My bad. Corrected.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 5 (20.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|2|00.00%|[3109, 3133]|
|:large_blue_circle:|com/starrocks/analysis/Analyzer.java|1|3|33.33%|[339, 340]|

File: be/src/service/internal_service.cpp [hunk 4]
```diff
@@ -793,21 +793,31 @@ void PInternalServiceImplBase<T>::get_file_schema(google::protobuf::RpcControlle
...
+        st = Status::InvalidArgument(err_msg);
         return;
     }
 
     DeferOp defer2([&p_scanner] { p_scanner->close(); });
 
     std::vector<SlotDescriptor> schema;
     st = p_scanner->get_schema(&schema);
+    if (!st.ok()) {
+        auto err_msg = fmt::format("get file schema failed, file path: {}, err: {}", scan_range.ranges[0].path, st);
```
Comment (chaoyli) [inline be/src/service/internal_service.cpp:817]: `st.get_error_msg()` is enough

Comment (rickif) [reply be/src/service/internal_service.cpp]: Updated.

**Comment by rickif** [comment]
@mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 5 (20.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|2|00.00%|[3109, 3133]|
|:large_blue_circle:|com/starrocks/analysis/Analyzer.java|1|3|33.33%|[339, 340]|

**Review by chaoyli — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**